homeassistant/components/overkiz/climate_entities/atlantic_pass_apc_heating_and_cooling_zone.py

```diff
@@ -0,0 +1,207 @@
+from ..entity import OverkizEntity
+
+OVERKIZ_TO_HVAC_MODE: dict[str, str] = {
+    OverkizCommandParam.AUTO: HVACMode.AUTO,
+    OverkizCommandParam.ECO: HVACMode.AUTO,
+    OverkizCommandParam.MANU: HVACMode.HEAT,
+    OverkizCommandParam.HEATING: HVACMode.HEAT,
+    OverkizCommandParam.STOP: HVACMode.OFF,
+    OverkizCommandParam.INTERNAL_SCHEDULING: HVACMode.AUTO,
+    OverkizCommandParam.COMFORT: HVACMode.HEAT,
+}
+
+HVAC_MODE_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_HVAC_MODE.items()}
+
+OVERKIZ_TO_PRESET_MODES: dict[str, str] = {
+    OverkizCommandParam.OFF: PRESET_ECO,
+    OverkizCommandParam.STOP: PRESET_ECO,
+    OverkizCommandParam.MANU: PRESET_COMFORT,
+    OverkizCommandParam.COMFORT: PRESET_COMFORT,
+    OverkizCommandParam.ABSENCE: PRESET_AWAY,
+    OverkizCommandParam.ECO: PRESET_ECO,
+    OverkizCommandParam.INTERNAL_SCHEDULING: PRESET_HOME,
+}
+
+PRESET_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODES.items()}
+
+OVERKIZ_TO_PROFILE_MODES: dict[str, str] = {
+    OverkizCommandParam.OFF: PRESET_SLEEP,
+    OverkizCommandParam.STOP: PRESET_SLEEP,
+    OverkizCommandParam.ECO: PRESET_ECO,
+    OverkizCommandParam.ABSENCE: PRESET_AWAY,
+    OverkizCommandParam.MANU: PRESET_COMFORT,
+    OverkizCommandParam.DEROGATION: PRESET_COMFORT,
+    OverkizCommandParam.COMFORT: PRESET_COMFORT,
+}
+
+
+class AtlanticPassAPCHeatingAndCoolingZone(OverkizEntity, ClimateEntity):
+    """Representation of Atlantic Pass APC Heating And Cooling Zone Zone Control."""
+
+    _attr_hvac_modes = [*HVAC_MODE_TO_OVERKIZ]
+    _attr_preset_modes = [*PRESET_MODES_TO_OVERKIZ]
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(
+            int(self.device_url.split("#", 1)[1]) + 1
+        )
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        if temperature := self.temperature_device.states[OverkizState.CORE_TEMPERATURE]:
+            return cast(float, temperature.value)
+
+        return None
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        return OVERKIZ_TO_HVAC_MODE[
+            cast(str, self.executor.select_state(OverkizState.IO_PASS_APC_HEATING_MODE))
+        ]
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        commands: list[list | str] = [
+            [
+                OverkizCommand.SET_PASS_APC_HEATING_MODE,
+                [HVAC_MODE_TO_OVERKIZ[hvac_mode]],
+            ],
+            # We also needs to execute these 3 commands to make it work correctly
```

Comment:
Is it an idea to split this PR in two? So add the commands just in a row, like we did in the past and do in other implementations.

And work in a separate PR on executing multiple commands in one execution? This will speed up the approval of this PR and we can have a more in detail discussion about the change. Now we have 2 big changes in one PR.

https://github.com/home-assistant/core/pull/76690/files#diff-89e3e126bd14fe1425aef65432a1076905a7bc0dd26b3b2e9fa29f8d7a5cade4R53-R63

Reply:
Done as you requested here PR #79576
Closing this thread.